### PR TITLE
docs,packages/dd-trace: add dogstatsd.hostname option

### PR DESCRIPTION
**Pull request by maxenglander**
### What does this PR do?

This PR adds a `dogstatsd.hostname` option to the tracer
`init()` function. When supplied, it will be used as the
destination for DogStatsD metrics generated by dd-trace-js.
If not supplied, the current behavior is used, which is to send
DogStatsD metrics to `hostname`.

### Motivation

dd-trace-js does not currently support sending DogStatsD metrics
over Unix Domain Sockets. Until that is supported, it would be
helpful to allow sending DogStatsD metrics to a host other than
then the APM endpoint. This would allow DogStatsD metrics to be
proxied, for example via socat running in the same container as
the application.

### Plugin Checklist

- [ ] Unit tests.
- [ ] TypeScript [definitions][1].
- [ ] TypeScript [tests][2].
- [ ] API [documentation][3].
- [ ] CircleCI [jobs/workflows][4].
- [ ] Plugin is [exported][5].

[1]: https://github.com/DataDog/dd-trace-js/blob/master/index.d.ts
[2]: https://github.com/DataDog/dd-trace-js/blob/master/docs/test.ts
[3]: https://github.com/DataDog/dd-trace-js/blob/master/docs/API.md
[4]: https://github.com/DataDog/dd-trace-js/blob/master/.circleci/config.yml
[5]: https://github.com/DataDog/dd-trace-js/blob/master/packages/dd-trace/src/plugins/index.js

### Additional Notes

This is a workaround, but not a solution, for #993.

### CODE REVIEW

File: docs/API.md [hunk 1]
```diff
@@ -420,6 +420,7 @@ Options can be configured as a parameter to the [init()](./interfaces/tracer.htm
 | url            | `DD_TRACE_AGENT_URL`           | -           | The url of the trace agent that the tracer will submit to. Takes priority over hostname and port, if set. |
 | hostname       | `DD_TRACE_AGENT_HOSTNAME`      | `localhost` | The address of the agent that the tracer will submit to. |
 | port           | `DD_TRACE_AGENT_PORT`          | `8126`      | The port of the trace agent that the tracer will submit to. |
+| dogstatsd.hostname | -                          | -           | The address of the Dogstatsd agent that metrics will be submitted to. Falls back to `hostname`. |
```
Comment (rochdev) [inline docs/API.md:423]: Let's add an environment variable as well to be consistent with the port.

Comment (rochdev) [inline docs/API.md:423]:
```suggestion
| dogstatsd.hostname | -                          | -           | The address of the Dogstatsd agent that metrics will be submitted to. Falls back to the same value as the `hostname` option. |
```

File: packages/dd-trace/test/platform/node/index.spec.js [hunk 2]
```diff
@@ -381,8 +381,8 @@ describe('Platform', () => {
 
         platform = {
           _config: {
-            hostname: 'localhost',
```
Comment (rochdev) [inline packages/dd-trace/test/platform/node/index.spec.js:384]: Why was this removed? I'm also not sure why removing it didn't fail any test.

File: packages/dd-trace/test/config.spec.js [hunk 3]
```diff
@@ -104,6 +104,10 @@ describe('Config', () => {
 
     expect(config).to.have.property('enabled', false)
     expect(config).to.have.property('debug', true)
+    // dogstatsd.hostname is never set to trace agent URL, since this can
```
Comment (rochdev) [inline packages/dd-trace/test/config.spec.js:107]: This should be a test instead of a comment.

File: packages/dd-trace/test/platform/node/index.spec.js (hunk 2 above)
Comment (maxenglander) [reply packages/dd-trace/test/platform/node/index.spec.js:384]: I've updated `platform/node/metrics.js` to use `config.dogstatsd.hostname` instead of `config.hostname`. 

https://github.com/DataDog/dd-trace-js/pull/995/files#diff-ded06582bc11034346decd1b6b2bcbbbR52

Tests are passing because the same value that was being passed in as `config.hostname` in tests is now being passed in as `config.dogstatsd.hostname`, which is set to the value of `config.hostname` when no value is explicitly set by the user.

**Comment by maxenglander** [comment]
All good -- added a type constraint and a change to `docs/test.js` (although I'm not entirely clear how the test works).
